testdata/default/cheats/Wallet.t.sol

```diff
@@ -104,10 +104,36 @@ contract WalletTest is DSTest {
         assertEq(recovered, wallet.addr);
     }
 
+    function testSignEIP2098WithWalletDigest(uint256 pkSeed, bytes32 digest) public {
+        uint256 pk = bound(pkSeed, 1, Q - 1);
+
+        Vm.Wallet memory wallet = vm.createWallet(pk);
+
+        (bytes32 r, bytes32 vs) = vm.signEIP2098(wallet, digest);
+
+        // Extract `s` from `vs`.
+        // The mask 0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff has all bits set to 1 except the leftmost bit, which is 0.
+        // Performing a bitwise AND operation with this mask clears the leftmost bit of `vs`, giving us `s`.
+        bytes32 s = bytes32(uint256(vs) & 0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
```

Comment:
nit: this can be simplified to `((x << 1) >> 1)` for readability

Reply:
done in [chore: update signCompact tests](https://github.com/foundry-rs/foundry/pull/8538/commits/e402b44a6266b03a9461f96e8464d466910b8c87)